workspaces/cogito-react-ui/source/__mocks__/qrcode.react.js

```diff
@@ -0,0 +1,5 @@
+import React from 'react'
+
+const MockedQRCode = ({ value }) => <div>{ value }</div>
+
+export default MockedQRCode
```

Comment:
Mocking components is best avoided in general as this makes tests more coupled to the implementation details. The reason you mocked this component here is probably inability of `jest` to deal with canvas. We have already tests that face this problem, and the general solution for that is to use `jest-canvas-mock` package. You also normally need to add `jest.config.js` and `setup-tests.js` file the same way it is done in the `demo-app` workspace.

Initially it was only needed for the demo-app, as other ui-related packages were not under test. Now, we cover most of them and so we started to observe some duplication. I was planning to improve that, but I prefer to wait for the work on web-sockets notifications to be merged to the master.

Reply:
We tried using the `jest-canvas-mock`, but couldn't verify the url being used in the canvas QR Code